DevelopmentTaskWorkflow.md

```diff
@@ -0,0 +1,44 @@
+Development task workflow
+=====================
+
+(for features, defects, etc. performed by the development team.)
+
+In general, we follow the [Github Flow](http://scottchacon.com/2011/08/31/github-flow.html).  The following provides some additional clarifications on our flavor of the workflow:
+
+1.  For each development task, a team member creates a [tickets](Tickets.md).  The ticket is assigned to a developer and optionally to a milestone.
+2.  For each ticket, the developer creates a feature branch.  The feature branch should be named `t<ticket #>-<short description with words - separated>`, e.g., `t286-export-stream`.  For example, to create and checkout a feature branch:
+
+        git checkout master
+        git checkout -b t286-export-stream
+
+    > Note:  In general, 1 development task = 1 ticket = 1 feature branch.  However, where appropriate, variations are acceptable.  In particular, a development task may be broken up into multiple tickets or multiple related tickets may be included in a single feature branch.
+
+3.  Developer does work and commits to feature branch.  The developer is welcome to commit as often as necessary for development purposes.  A commit to a feature branch may result in the code being "broken".  Each commit must be accompanied by a descriptive message.
+
+    Periodically, the developer should get the latest commits from master using rebase:
+    
+        git checkout mybranch
+        git rebase master
+        
+4.  When completed with the task or wanting to collaborate with other team members, the developer pushes the branch to origin.
+
+        git push -u origin mybranch
```

Comment:
From http://git-scm.com/docs/git-push:
>-u
--set-upstream
For every branch that is up to date or successfully pushed, add upstream (tracking) reference, used by argument-less git-pull[1] and other commands. For more information, see branch.<name>.merge in git-config[1].

From http://git-scm.com/docs/git-config:
branch.<name>.merge
>Defines, together with branch.<name>.remote, the upstream branch for the given branch. It tells git fetch/git pull/git rebase which branch to merge and can also affect git push (see push.default). When in branch <name>, it tells git fetch the default refspec to be marked for merging in FETCH_HEAD. The value is handled like the remote part of a refspec, and must match a ref which is fetched from the remote given by "branch.<name>.remote". The merge information is used by git pull (which at first calls git fetch) to lookup the default branch for merging. Without this option, git pull defaults to merge the first refspec fetched. Specify multiple values to get an octopus merge. If you wish to setup git pull so that it merges into <name> from another branch in the local repository, you can point branch.<name>.merge to the desired branch, and use the relative path setting . (a period) for branch.<name>.remote.